packages/react-core/src/components/TextInput/TextInput.tsx

```diff
@@ -60,6 +60,12 @@ export interface TextInputProps
   onFocus?: (event?: any) => void;
   /** Callback function when input is blurred (focus leaves) */
   onBlur?: (event?: any) => void;
+  /** icon variant */
+  iconVariant?: 'calendar' | 'clock';
+  /** Custom icon url to set as the input's background-image */
+  customIconUrl?: string;
```

Comment:
could this work with react-icons?

Reply:
I do not believe so, because it modifies the css background image which needs a url string... @mcoker does that check out?

Reply:
That's right, all of the icons in text input, form select (the down arrow), and text area are background images, so you can use anything you could pass to `background-image: url();`. If there were a utility to convert a react icon to a data URI, you could...